taipy/core/scenario/_scenario_manager.py

```diff
@@ -521,3 +521,47 @@ def _get_by_config_id(cls, config_id: str, version_number: Optional[str] = None)
         for fil in filters:
             fil.update({"config_id": config_id})
         return cls._repository._load_all(filters)
+
+    @classmethod
+    def _clone(cls, scenario: Scenario) -> Scenario:
+        """
+        Clone a scenario.
+
+        Arguments:
+            scenario (Scenario): The scenario to clone.
+
+        Returns:
+            Scenario: The cloned scenario.
+        """
+        cloned_scenario = cls._get(scenario)
+        cloned_scenario_id = cloned_scenario._new_id(cloned_scenario.config_id)
+        cloned_scenario.id = cloned_scenario_id
+        # TODO: update sequences
+
+        # Clone tasks and data nodes
+        _task_manager = _TaskManagerFactory._build_manager()
+        _data_manager = _DataManagerFactory._build_manager()
+
+        cloned_tasks = set()
+        for task in cloned_scenario.tasks.values():
+            cloned_tasks.add(_task_manager._clone(task, None, cloned_scenario_id))
+        cloned_scenario._tasks = cloned_tasks
```

Comment:
Duplicating tasks should depend on minimal scope of inputs and outputs data nodes.